### ISSUE TITLE
Cannot serialize Queue

### ISSUE THREAD

**Issue by chengscott** [fmfn/BayesianOptimization#370]
The following code works with `bayesian-optimization==1.2.0`, but fails with `bayesian-optimization==1.3.0` (or the latest one).

```python
import ray
from ray import tune
from ray.tune.search import ConcurrencyLimiter
from ray.tune.search.bayesopt import BayesOptSearch


def scoreFn(config):
    print(config)
    tune.report(
        ret=config['x'] * 1.2 + config['y'] * 1.2
    )

bo = BayesOptSearch(
    space={
        'x': (0.2, 0.95),
        'y': (0.5, 2.5),
    },
    metric='ret',
    mode='min',
)

tuner = tune.Tuner(
    scoreFn,
    tune_config=tune.TuneConfig(
        num_samples=24,
        search_alg=ConcurrencyLimiter(bo, max_concurrent=4),
    ),
)
results = tuner.fit()
```
The error in `bayesian-optimization==1.3.0`:
```shell
Traceback (most recent call last):
  File "/run/media/adam/Data/search_bo.py", line 38, in <module>
    tuner = tune.Tuner(
  File "/run/media/adam/Data/venv-py310/lib/python3.10/site-packages/ray/tune/tuner.py", line 152, in __init__
    self._local_tuner = TunerInternal(**kwargs)
  File "/run/media/adam/Data/venv-py310/lib/python3.10/site-packages/ray/tune/impl/tuner_internal.py", line 123, in __init__
    pickle.dump(self, fp)
  File "/run/media/adam/Data/venv-py310/lib/python3.10/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 57, in dump
    ).dump(obj)
  File "/run/media/adam/Data/venv-py310/lib/python3.10/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 627, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle '_thread.lock' object
```
After bisecting, the code works if I revert [commit 8cc8d9b](https://github.com/fmfn/BayesianOptimization/commit/8cc8d9bffd38cd8c5e7597f7ce92e1c9810d64fc).
A quick fix would be reverting that commit in the master branch.

**Comment by bwheelz36** [comment]
thanks for reporting. will fix ASAP - will also add some new test cases since that commit did pass all existing tests.

**Comment by bwheelz36** [comment]
hi, can you provide instructions for installing the libraries necessary to run this example?
I have tried every combination of `pip install ray` [I can find](https://docs.ray.io/en/latest/ray-overview/installation.html#installation) and I still cannot import `ray.tune.search`

**Comment by chengscott** [comment]
Bug reproduce in colab:
https://colab.research.google.com/drive/1UfjPGTY-9s5x_pnODy4X-EYpaKH_kgE6?usp=sharing

**Comment by cosmicBboy** [comment]
can confirm this issue comes up when trying to pickle the `BayesianOptimization` object:

```python
import pickle
from bayes_opt import BayesianOptimization

optimizer = BayesianOptimization(
    f=None,
    pbounds={"x": (-2, 2), "y": (-3, 3)},
    verbose=2,
    random_state=1,
)

pickle.dump(optimizer, open('p', 'w'))
```

**Comment by bwheelz36** [comment]
OK @chengscott - I think may ray issues were I started with python 3.11 which wouldn't install ray at all, then I went to 3.6 which installed but but must have been 1.x. With 3.10 it worked fine.

I've done what you suggested and reverted back to the old Queue implementation. I have also added a test based on @cosmicBboy's code to ensure that the the BayesianOptimization can be serialized. I will release this under 1.3.1 shortly.
